lib/publiccloud/utils.pm

```diff
@@ -885,4 +886,112 @@ sub upload_asset_on_remote {
 }
 
+
+=head2 zypper_call_remote
+
+zypper_call_remote($instance, $cmd, %args)
+
+Remote package management.
+To run on a remote public cloud vm instance a non-interative system manager
+zypper or a transactional-update package command (depending on 'DISTRI'),
+accepting in input the subcommand, then composing the full command to run.
+Free command input also allowed, skipping pre-composition,
+  when 'zypper' or 'transactional-update' already present in input;
+  useful when needing to run a complex one-liner or pipeline package management command.
+
+Parameters:
+  $instance: pointer to publiccloud instance class;
+  $cmd|$args{cmd}: default:'update'; subcommand of zypper or tr-up package management to run
+    OR free full command line, including zypper/tr-up command;
+  %args: dictionary of the arguments defined in input, like: (argument => value, ...):
+    $args{log}: full-path-filename, to save output for upload;
+    $args{global_options}: global options before subcommand, default '-n', as by documentation;
+    $args{environment}: eventual in-line shell pre-setting (i.e. "LANG=C zypper lr ...");
+    $args{proceed_on_failure}: default=0: works as assertion; 1: doesn't stop on error.
+    $args{retry}: to trigger remote retries on error, by default (0) inactve.
+    Included all the $args{...} of run_ssh_command()/retry_ssh_command(), form lib/publiccloud/instance.pm.
+
+Return, same as run_ssh_command(): rc_only=1 [default], as script_run; rc_only=0, as script_output.
+
+=cut
+
+sub zypper_call_remote {
+    my $instance = shift;
+    my %args = testapi::compat_args({cmd => undef}, ['cmd'], @_);
+    return unless ($instance);
+
+    # preset
+    $args{rc_only} //= 1;
+    $args{retry} //= 0;
+    $args{proceed_on_failure} //= 0;
+    $args{global_options} //= '-n';
+    $args{environment} //= '';
+    my $log = $args{log};
+    my $printer = ($log) ? "| tee $log; ( exit \${PIPESTATUS[0]} )" : '';
+    my $subcmd = $args{cmd};
+    my $command;
+    my $ret;
+
+    # compose command
+    # accept full command line (expert) entry, skipping next helper pre-composition,
+    $command = $subcmd if ($subcmd =~ /((^|\b)transactional-update\b)(?![-\/])/ || $subcmd =~ /((^|\b)zypper\b)(?![-\/])/);
+    # compose tran-up command
+    $command = zypper_trup_composer(%args) if (!$command && is_transactional);
+    # compose zypper command; default 'update'
+    $subcmd //= 'up';
+    $command = "sudo $args{environment} zypper $args{global_options} " . $subcmd unless ($command);
+    # cmd setup
+    $args{cmd} = $command . $printer;
+    # remove extra-args
+    delete($args{environment});
+    delete($args{global_option});
+    delete($args{log});
+
+    # run command
+    diag("zypper_call_remote: $command\n args: " . Dumper(%args));
+    if ($args{retry} > 1) {
+        $ret = $instance->retry_ssh_command(%args);
+    } else {
+        $ret = $instance->run_ssh_command(%args);
+    }
+
+    # result
+    if ($args{rc_only}) {
```

Comment:
I would expect that `rc_only` should not be involved in deciding whether function failing or not